### PR TITLE
Support overriding ROS_DISTRO for ROS2 ros_environment

### PR DESCRIPTION
**Pull request by paulbovbel**
Provides the same functionality as available with ROS1 ros_environment

### CODE REVIEW

**Review by mikaelarguedas — APPROVED** [review]
:+1:

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,11 @@ set(ROS_VERSION "2")
 set(ROS_DISTRO "bouncy")
 set(ROS_PYTHON_VERSION "3")
 
+# allow overriding the distro name
+if(NOT $ENV{ROS_DISTRO_OVERRIDE} STREQUAL "")
```
Comment (dirk-thomas) [inline CMakeLists.txt]: This line results in a warning when CMake is invoked with `--warn-uninitialized`. See https://github.com/ros2/rosidl_typesupport_connext/blob/e9d8da960abc691b7f304fabd256d97d006281ee/rosidl_typesupport_connext_cpp/CMakeLists.txt#L5-L6 for an example how to check if the environment variable is defined.

Comment (mikaelarguedas) [reply CMakeLists.txt]: What is the policy you want to apply to this repository ?

Comply with more rigid CMake flags only on the bouncy branch and onward? or update everywhere?

Comment (dirk-thomas) [reply CMakeLists.txt]: I don't mind if it is backported. As long as the lastest development branch has the fix.

Comment (paulbovbel) [reply CMakeLists.txt]: Fixed, and side-ported to melodic https://github.com/ros/ros_environment/pull/11

Comment (dirk-thomas) [reply CMakeLists.txt]: Thanks.

**Pull request merged**